src/asl.ts

```diff
@@ -1714,11 +1682,55 @@ export class ASL {
             },
           };
         });
+      } else if (expr.op === "=") {
+        if (!(isVariableReference(expr.left) || isIdentifier(expr.left))) {
+          throw new SynthError(
+            ErrorCodes.Unexpected_Error,
+            "Expected left side of assignment to be a variable."
+          );
+        }
+        return this.evalContext(expr, (evalExpr) => {
+          const right = evalExpr(expr.right);
+          const left = evalExpr(expr.left);
+
+          if (!ASLGraph.isJsonPath(left)) {
+            throw new SynthError(
+              ErrorCodes.Unexpected_Error,
+              `Expected assignment to target a variable, found: ${left.value}`
+            );
+          }
+
+          return {
+            ...ASLGraph.passWithInput(
+              {
+                Type: "Pass",
+                ResultPath: left.jsonPath,
+                Next: ASLGraph.DeferNext,
+              },
+              right
+            ),
+            output: left,
+          };
+        });
+      } else if (
+        expr.op === "+" ||
+        expr.op === "-" ||
+        expr.op === "*" ||
+        expr.op === "/" ||
+        expr.op === "%" ||
+        expr.op === "+=" ||
+        expr.op === "-=" ||
+        expr.op === "*=" ||
+        expr.op === "/=" ||
+        expr.op === "%="
+      ) {
+        // TODO: support string concat?
```

Comment:
can be done with `States.Format`? Track in issue?

Reply:
It can, just the awkwardness of supporting `+` for one use case.

Reply:
And possibly some complexity around knowing the types at compile time.